src/main/java/graphql/incremental/IncrementalExecutionResultImpl.java

```diff
@@ -0,0 +1,103 @@
+package graphql.incremental;
+
+import graphql.ExecutionResult;
+import graphql.ExecutionResultImpl;
+import graphql.ExperimentalApi;
+import org.reactivestreams.Publisher;
+
+import javax.annotation.Nullable;
+import java.util.LinkedHashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static java.util.stream.Collectors.toList;
+
+@ExperimentalApi
+public class IncrementalExecutionResultImpl extends ExecutionResultImpl implements IncrementalExecutionResult {
```

Comment:
This shows a hole in how we did subscriptions

in subscriptions we do

```
Publisher<ExecutionResult> msgStream = executionResult.getData()
```

that is the ER data holds the publisher.

This is better since it has more attrs and it is more complex - but now subscriptions makes me sad

Reply:
> since it has more attrs 
Yep. The different shape of the results really required a whole new class hierarchy. 
I'm glad you liked it 👍

Reply:
I am not 100% a hierarchy is what we want here , but I could be convinced.

Reply:
Are you now convinced - we talked in the zoom call